includes/html/pages/device/apps/suricata.inc.php

```diff
@@ -1,45 +1,824 @@
+        'decoder' => 'Decoder', '#1.1' => '(', 'decoder_erspan' => 'ERSPAN', 'decoder_gre' => 'GRE', 'decoder_icmpv4' => 'ICMPv4', 'decoder_icmpv6' => 'ICMPv6',
+        'decoder_ipv4' => 'IPv4', 'decoder_ipv6' => 'IPv6', 'decoder_ltnull' => 'LT Null', 'decoder_mpls' => 'MPLS', 'decoder_nsh' => 'NSH', 'decoder_ppp' => 'PPP',
+        'decoder_pppoe' => 'PPPoE', 'decoder_tcp' => 'TCP', 'decoder_udp' => 'UDP', 'decoder_vlan' => 'VLAN', 'decoder_vntag' => 'VNTag', '#3' => ')', ];
+
+    $suricata_pages_no_comma = ['errors' => 1, 'errors_parser' => 1, 'memuse' => 1, 'memuse_details' => 1, 'applayer' => 1, 'applayer_tx' => 1, 'decoder' => 1, 'decoder_vntag' => 1];
+
+    $page_count = 0;
+    foreach ($suricata_pages as $page => $page_description) {
+        if (preg_match('/^\#/', $page)) {
+            echo $page_description;
+        } else {
+            $label = $vars['suricata_graph_set'] == $page
+                ? '<span class="pagemenu-selected">' . $page_description . '</span>'
+                : $page_description;
+
+            if (isset($vars['sinstance'])) {
+                echo generate_link($label, $link_array, ['sinstance' => $vars['sinstance'], 'suricata_graph_set' => $page]);
+            } else {
+                echo generate_link($label, $link_array, ['suricata_graph_set' => $page]);
+            }
+
+            if ($page_count < (count($suricata_pages) - 1) && ! isset($suricata_pages_no_comma[$page])) {
+                echo ', ';
+            }
+        }
+        $page_count++;
+    }
+}
+
 print_optionbar_end();
 
-$graphs = [
-    'suricata_packets' => 'Packets',
-    'suricata_bytes' => 'Bytes',
-    'suricata_nasty_delta' => 'Drops or Errors Delta',
-    'suricata_nasty_percent' => 'Drops or Errors Percent',
-    'suricata_dec_proto' => 'Decoder Protocols',
-    'suricata_flow_proto' => 'Flow Protocols',
-    'suricata_app_flows' => 'App Layer Flows',
-    'suricata_app_tx' => 'App Layer TX',
-    'suricata_mem_use' => 'Memory Usage',
-    'suricata_uptime' => 'Uptime',
-    'suricata_alert' => 'Alert Status',
-];
+if ($app_data['version'] == 1) {
+    $graphs = [
+        'suricata_packets' => 'Packets',
+        'suricata_bytes' => 'Bytes',
+        'suricata_nasty_delta' => 'Drops or Errors Delta',
+        'suricata_nasty_percent' => 'Drops or Errors Percent',
+        'suricata_dec_proto' => 'Decoder Protocols',
+        'suricata_flow_proto' => 'Flow Protocols',
+        'suricata_app_flows' => 'App Layer Flows',
+        'suricata_app_tx' => 'App Layer TX',
+        'suricata_mem_use' => 'Memory Usage',
+        'suricata_uptime' => 'Uptime',
+        'suricata_alert' => 'Alert Status',
+    ];
+} elseif ($app_data['version'] == 2) {
+    if (! isset($vars['suricata_graph_set'])) {
+        $vars['suricata_graph_set'] = 'general';
+    }
+
+    if (strcmp($vars['suricata_graph_set'], 'general') == 0) {
```

Comment:
kind of weird using strcmp() when you are just checking if they are equal

Reply:
Derp! Good point. `==` is way better here. I tend to forget it works for this as well and strcmp is what comes up in the docs when I forget. :P

Reply:
And fixed!